src/futils.cpp

```diff
@@ -101,52 +105,43 @@ namespace Exiv2 {
         return isdigit(ch) ? ch - '0' : tolower(ch) - 'a' + 10;
     }
 
-    std::string urlencode(const char* str) {
-        const char* pstr = str;
-        // \todo try to use std::string for buf and avoid the creation of another string for just
-        // returning the final value
-        auto buf = new char[strlen(str) * 3 + 1];
-        char* pbuf = buf;
-        while (*pstr) {
-            if (isalnum(*pstr) || *pstr == '-' || *pstr == '_' || *pstr == '.' || *pstr == '~')
-                *pbuf++ = *pstr;
-            else if (*pstr == ' ')
-                *pbuf++ = '+';
-            else
-                *pbuf++ = '%', *pbuf++ = to_hex(*pstr >> 4), *pbuf++ = to_hex(*pstr & 15);
-            pstr++;
+    std::string urlencode(const std::string_view& str)
+    {
+        std::string encoded;
+        encoded.reserve(str.size() * 3);
+        for (auto c : str) {
```

Comment:
```suggestion
        for (auto x : str) {
            const uint8_t c = static_cast<uint8_t>(x);
```
(To fix the CodeQL warning below.)